src/iosrv/drpc_client.c

```diff
@@ -38,21 +40,49 @@ struct drpc *dss_drpc_ctx;
 static int
 notify_ready(void)
 {
-	Drpc__Call     *req;
-	Drpc__Response *resp;
-	int		rc;
-
-	req = drpc_call_create(dss_drpc_ctx,
-			       DRPC_MODULE_SRV,
-			       DRPC_METHOD_SRV_NOTIFY_READY);
-	if (req == NULL) {
-		return -DER_NOMEM;
+	Srv__NotifyReadyReq	req = SRV__NOTIFY_READY_REQ__INIT;
+	uint8_t		       *reqb;
+	size_t			reqb_size;
+	Drpc__Call	       *dreq;
+	Drpc__Response	       *dresp;
+	int			rc;
+
+	rc = crt_self_uri_get(0 /* tag */, &req.uri);
+	if (rc != 0)
+		goto out;
+	req.nctxs = DSS_CTX_NR_TOTAL;
+
+	reqb_size = srv__notify_ready_req__get_packed_size(&req);
+	D_ALLOC(reqb, reqb_size);
+	if (reqb == NULL) {
+		rc = -DER_NOMEM;
+		goto out_uri;
 	}
+	srv__notify_ready_req__pack(&req, reqb);
 
-	rc = drpc_call(dss_drpc_ctx, R_SYNC, req, &resp);
+	dreq = drpc_call_create(dss_drpc_ctx, DRPC_MODULE_SRV,
+				DRPC_METHOD_SRV_NOTIFY_READY);
+	if (dreq == NULL) {
+		rc = -DER_NOMEM;
+		D_FREE(reqb);
+		goto out_uri;
+	}
+	dreq->module = DRPC_MODULE_SRV;
+	dreq->method = DRPC_METHOD_SRV_NOTIFY_READY;
+	dreq->body.len = reqb_size;
+	dreq->body.data = reqb;
 
-	drpc_call_free(req);
-	drpc_response_free(resp);
+	rc = drpc_call(dss_drpc_ctx, R_SYNC, dreq, &dresp);
+	if (rc != 0)
+		goto out_dreq;
+
+	drpc_response_free(dresp);
```

Comment:
Just a bit curious, though I know this is unchanged from the old logic... Would there be any value in inspecting the response before freeing it? I suppose the most we could do is log an error if we got a bad response.

Reply:
I can think of "memory corruptions" or "the dRPC Go client encounters an error but doesn't report it" as the only two reasons why a check here would be useful. But we can add a status check anyway, as it shouldn't be just two lines of code.